pkg/pulumiyaml/analyser_test.go

```diff
@@ -367,3 +367,22 @@ func TestConfigCompatibility(t *testing.T) {
 		})
 	}
 }
+
+func TestNonStringKeyInObjectReturnsError(t *testing.T) {
+	t.Parallel()
+
+	tc := typeCache{
+		exprs: make(map[ast.Expr]schema.Type),
+	}
+	expr := &ast.ObjectExpr{
+		Entries: []ast.ObjectProperty{
+			{
+				Key: &ast.BooleanExpr{},
+			},
+		},
+	}
+	_ = tc.typeExpr(nil, expr)
+	require.Equal(t, 1, len(tc.exprs))
+	require.Equal(t, "Object key must be a string, got *ast.BooleanExpr",
```

Comment:
That's not a friendly error for users. Can we get the user-friendly type of the expression instead? So it would just say "got boolean" here.

Reply:
Not easily I think.  We do the same thing in other parts of the file: https://github.com/pulumi/pulumi-yaml/pull/816/files#diff-081a538c30d4b80890272ca7cc1aaa9b8e84fc6592814ab06452220440cc0eacR1095-R1108.  I guess we could add an extra method to the Expr interface like `Type()` for this.  Not sure it's worth it though.

Reply:
merge this because it fixes a panic, but we probably should make these errors nicer. I'll raise an issue about it.